### ISSUE TITLE
"Extract installed packages from a container image. This would be used for layered images."

### ISSUE THREAD

**Issue by cgwalters** [konflux-ci/rpm-lockfile-prototype#4]
> Extract installed packages from a container image. This would be used for layered images.

I think we should instead accept one of several declarative input formats.  This is discussed in https://github.com/containerbuildsystem/cachi2/discussions/465

Trying to extract after the fact just makes everything circular.

**Comment by cgwalters** [comment]
You’re right sorry.

However there is an alternative implementation: Add the machine-readable declarative input used to generate the base image into its metadata.

Today for UBI it’s a kickstart.  Then we just need the immutable yum repo versions used and that should be sufficient to know what is in the base.

(If the base switches to lock files as used in this prototype we could just stick that whole thing in metadata but - to connect with the other thread I think “immutable versioned repos” is far more ergonomic and readable than lock files.)

Of course in the general case this needs to support multiple layers, and changes(upgrade, downgrade, removal) from prior layers.

So the metadata key would need to be an array, a bit like the “history” metadata in the config.

**Comment by lubomir** [comment]
I completely agree that there are better ways to solve this problem as long as the base image is cooperating.

The current implementation is problematic on many fronts (hard to run in container, inefficient…).
